### PR TITLE
[baremetal] Add random delays to sha256 to protect against fault injection

### PR DESCRIPTION
**Pull request by AndrzejKurek**
Signed-off-by: Andrzej Kurek <andrzej.kurek@arm.com>

### CODE REVIEW

**Review by shelib01 — DISMISSED** [review]

File: library/aes.c [hunk 1]
```diff
@@ -1172,6 +1172,7 @@ int mbedtls_internal_aes_encrypt( mbedtls_aes_context *ctx,
     } while( stop_mark == 0 );
 
     // Calculate final AES round + dummy rounds
+    mbedtls_platform_random_delay();
```
Comment (shelib01) [inline library/aes.c:1175]: I thought about it again(during masking implementaion) , i think we shouldn't add it . Please remove

Comment (AndrzejKurek) [reply library/aes.c:1175]: Only this one?

Comment (shelib01) [reply library/aes.c:1175]: Only this one

**Comment by AndrzejKurek** [comment]
Removed one delay and rebased.

**Review by shelib01 — APPROVED** [review]

**Pull request merged**